src/plugins/discover/public/build_services.ts

```diff
@@ -72,7 +73,7 @@ export interface DiscoverServices {
   spaces?: SpacesApi;
 }
 
-export function buildServices(
+export const buildServices = memoize(function (
```

Comment:
`memoize` was added to optimise services initialisation for multiple saved search embeddables.

Reply:
The only concern here is, `lodash` bundle import on plugin start. We have already used somewhere `lodash` in the plugin initialisations, isn't it? Well, this import should not lead to unexpected increase of loading time.

Reply:
sounds good

Reply:
and works good